coremltools/coremltools/converters/mil/test_flexible_shape_inputs.py

```diff
@@ -0,0 +1,135 @@
+# Copyright (c) 2021, Apple Inc. All rights reserved.
+#
+# Use of this source code is governed by a BSD-3-clause license that can be
+# found in the LICENSE.txt file or at https://opensource.org/licenses/BSD-3-Clause
+
+import coremltools as ct
+from coremltools._deps import _HAS_TORCH, MSG_TORCH_NOT_FOUND
+import numpy as np
+import PIL.Image
+import pytest
+
+if _HAS_TORCH:
+    import torch
+
+    class TestConvModule(torch.nn.Module):
+        def __init__(self, in_channels=3, out_channels=10, kernel_size=3):
+            super(TestConvModule, self).__init__()
+            self.conv = torch.nn.Conv2d(in_channels, out_channels,
+                                        kernel_size)
+        def forward(self, x):
+            return self.conv(x)
+
+def _numpy_array_to_pil_image(x):
+    """
+    convert x of shape (1, 3, H, W) to PIL image
+    """
+    assert len(x.shape) == 4
+    assert list(x.shape[:2]) == [1, 3]
+    x = x[0, :, :, :] # (3, H, W)
+    x = np.transpose(x, [1, 2, 0]) # (H, W, 3)
+    x = x.astype(np.uint8)
+    return PIL.Image.fromarray(x)
+
+def _assert_torch_coreml_output_shapes(coreml_model, spec, torch_model, torch_example_input, is_image_input=False):
+    torch_out = torch_model(torch_example_input)
+    input_name = spec.description.input[0].name
+    output_name = spec.description.output[0].name
+    input_dict = {}
+    if is_image_input:
+        input_dict[input_name] = _numpy_array_to_pil_image(torch_example_input.numpy())
+    else:
+        input_dict[input_name] = torch_example_input.numpy()
+    coreml_out = coreml_model.predict(input_dict)[output_name]
+    assert torch_out.shape == coreml_out.shape
+
+
+@pytest.mark.skipif(not _HAS_TORCH or not ct.utils._is_macos(), reason=MSG_TORCH_NOT_FOUND)
+class TestFlexibleInputShapes:
+
+    @pytest.mark.parametrize("convert_to", ['neuralnetwork', 'mlprogram'])
+    def test_multiarray_input_rangedim(self, convert_to):
+        if convert_to == "mlprogram" and ct.utils._macos_version() < (12, 0):
+            return
+
+        example_input = torch.rand(1, 3, 50, 50)
```

Comment:
Might be good to set a random seed at the top of the file.

Reply:
Yeah that may be useful in general, although here in these tests, I'm not even comparing the output values between torch and coreml. Just making sure that the shapes match, since the tests are geared for flexible shape tests. 
Although, that said, I think its a good idea to add asserts for numerical values. I'll do that and then also set up the random seed!